### PR TITLE
Add mods for old VGC formats

### PR DESCRIPTION
**Pull request by Karthik99999**
~~I've added mods for VGC 11/12, VGC 14/15 will be done in this PR as well at some point~~

Check my comment below for more info

### CODE REVIEW

**Comment by KrisXV** [comment]
Whats the point of a vgc 11 mod when its just vgc 12 + 5 unreleased items, you could theoretically just ban those?

**Comment by Karthik99999** [comment]
@KrisXV 

Some notes:
- The following years have been added as new challenge-only formats:

| Year | Mod     |
|------|---------|
| 2009 | gen4pt  |
| 2010 | gen4    |
| 2011 | gen5bw1 |
| 2012 | gen5bw1  |
| 2014 | gen6xy  |

- We named the mods `gen4pt`, `gen5bw1` and `gen6xy`, which differs from the current mod in use for vgc 2017 (`vgc17`)
- There are some event moves that weren't released in 2011, but we considered them insignificant enough to not create a mod for 2011

**Review by DaWoblefet — APPROVED** [review]
I wouldn't be surprised if there's a few things we missed here or there, but Karthik did a great job with this. I'm excited to have proper oldgen VGC support!

File: data/mods/gen6xy/pokedex.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		unreleasedHidden: true,
+	},
+	amaura: {
+		inherit: true,
+		unreleasedHidden: true,
+	},
+	aurorus: {
+		inherit: true,
+		unreleasedHidden: true,
+	},
```
Comment (KrisXV) [inline data/mods/gen6xy/pokedex.ts:41]: Hidden ability gen 5 starters weren't released until ORAS came out, so they should probably be marked as unreleased in here

File: data/mods/gen6xy/moves.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+		isNonstandard: "Future",
+	},
+	originpulse: {
+		inherit: true,
+		isNonstandard: "Future",
+	},
+	precipiceblades: {
+		inherit: true,
+		isNonstandard: "Future",
+	},
```
Comment (KrisXV) [inline data/mods/gen6xy/moves.ts:17]: all 4 of the pokemon (and their items/abilities in the case of the primals/mray) affiliated with these moves should be marked as Future (or unreleased in hoopa-confined's case, unbound would be Future/Nonexistent)

**Review by KrisXV — CHANGES_REQUESTED** [review]
I didn't look over all of the details for the `gen6xy` mod and didn't go over the `gen5bw1` mod at all, but a lot of the stuff seems to be missing. A good way to think of gen6xy and gen5bw1 in the scope of PS is "Gen 5 before BW2 existed" and "Gen 6 before ORAS existed", or even "Gen 4 before HGSS existed" in the case of `gen4pt`.

File: data/mods/gen6xy/moves.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+};
```
Comment (Marty-D) [inline data/mods/gen6xy/moves.ts:18]: I believe Thousand Arrows/Waves need to be here as Unobtainable so Smeargle can't use them?

Comment (Karthik99999) [reply data/mods/gen6xy/moves.ts:18]: Both moves are already marked as `Unobtainable` in the base gen 6 mod

**Review by Marty-D — APPROVED** [review]
Looks good, thanks Karthik! 👍

**Pull request merged**